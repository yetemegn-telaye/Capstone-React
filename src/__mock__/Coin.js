import React from "react";
import { BrowserRouter, Link } from "react-router-dom";


const Coin = ()=>{
    const coinsArr = [{
          id: "bitcoin",
          coinName: "Bitcoin",
          rank: "1",
          symbol: "BTC",
          priceUsd: "16954.3049792354269760",
    }];
    return(
        <BrowserRouter>
        
        
        <div className="coins-container">
          {coinsArr.map((coin) => (
            <div key={coin.id} className="coin-container">
            <h4>
              {coin.coinName}
              {coin.symbol}
            </h4>
            <p>
              Rank:
              {coin.rank}
            </p>
            <p>
              Price:
              {parseFloat(coin.priceUsd).toFixed(3)}
            </p>
            <Link to={`/coin-details/${coin.id}`}>View</Link>
          </div>
          ))}
        </div>
    
        </BrowserRouter>
    )
}
export default Coin;