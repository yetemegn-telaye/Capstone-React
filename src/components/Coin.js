import React from 'react';

const Coin = () => (
  <div>
    Coin component
    <button type="button">Click to see coin details</button>
  </div>
);
export default Coin;
