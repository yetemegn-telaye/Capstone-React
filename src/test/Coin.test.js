import renderer from 'react-test-renderer';
import Coin from "../__mock__/Coin";



describe('Coin', ()=>{
    it('Coin renders correctly',()=>{
        const coin = renderer.create(<Coin />).toJSON();
        expect(coin).toMatchSnapshot();
    });
});