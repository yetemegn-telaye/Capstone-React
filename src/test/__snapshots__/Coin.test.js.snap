// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin Coin renders correctly 1`] = `
<div
  className="coins-container"
>
  <div
    className="coin-container"
  >
    <h4>
      Bitcoin
      BTC
    </h4>
    <p>
      Rank:
      1
    </p>
    <p>
      Price:
      16954.305
    </p>
    <a
      href="/coin-details/bitcoin"
      onClick={[Function]}
    >
      View
    </a>
  </div>
</div>
`;
